flink-autoscaler/src/main/java/org/apache/flink/autoscaler/config/AutoScalerOptions.java

```diff
@@ -72,6 +72,15 @@ private static ConfigOptions.OptionBuilder autoScalerConfig(String key) {
                     .withDescription(
                             "Stabilization period in which no new scaling will be executed");
 
+    public static final ConfigOption<List<String>> FORBID_PERIOD =
+            autoScalerConfig("forbid.periods")
+                    .stringType()
+                    .asList()
+                    .defaultValues()
+                    .withDeprecatedKeys(deprecatedOperatorConfigKey("forbid.periods"))
+                    .withDescription(
+                            "A (semicolon-separated) list of certain times of the day during which autoscaling is forbidden, 10:00:00-11:00:00;21:30:00-22:30:00 for example");
```

Comment:
Where did you get this syntax from? What are the potential alternatives?

Reply:
This was also my first thought seeing this. Ideally, we would use some available standard for time spans. I think this also has to include the time zone.

What about different times at different days of the week / month?

Reply:
I've seached several time scheduler framework, `Quartz Scheduler` seems to be feasible.
1. It supports `certain time of day` `certain days of the week` `certain days of the month` , etc.
https://github.com/quartz-scheduler/quartz/blob/main/docs/introduction.adoc#job-scheduling
2. Here is the implementation: 
https://github.com/quartz-scheduler/quartz/blob/a5c4d27e963f51097f9b2777489d310a88897ca4/quartz/src/main/java/org/quartz/Calendar.java#L21-L41
https://github.com/quartz-scheduler/quartz/blob/a5c4d27e963f51097f9b2777489d310a88897ca4/quartz/src/main/java/org/quartz/impl/calendar/CronCalendar.java#L10-L26

Reply:
Sounds good, we already use Quartz for `periodic.checkpoint.interval`.

Reply:
I checked the pr https://github.com/apache/flink-kubernetes-operator/pull/656 which introduced `class CronExpression` from log4j-core and log4j-core copied from Quartz.

1.  So can we set the `excluded periods` config to be a cron expression string and use it's `isSatisfiedBy(final Date date)` method to judge whether in excluded periods
2.  Since standardized cron string doesn't contain a timeZone, should we just use the default timeZone or some other way to pass it in?

Reply:
Yes, if the cron-based notation allows time ranges that would be perfect. We could just use UTC and document this for now.

Reply:
Quartz's  DailyCalendar along with CronCalendar allows  `daily excluded, days of the week/month excluded, etc.`
I've updated the config and implemented it, looking forward for the review.